### PR TITLE
Property DDS perf improvements 2

### PR DESCRIPTION
**Pull request by ruiterr**
More performance improvements, for the propertyDDS:

* Use msgpackr for binary serialization
* Precompile creation functions in the PropertyFactory.

@evaliyev PTAL

### CODE REVIEW

File: experimental/PropertyDDS/packages/property-properties/src/propertyFactory.js [hunk 1]
```diff
@@ -966,36 +975,401 @@ class PropertyFactory {
...
+                    creationStack.push({
+                        parent: property,
+                        id: id,
+                        entry: child,
+                        setGuid: currentEntry.entry.assignGuid && id ===
+                        'guid'
+                    })
+                }
+            } else {
+                // If there are nor children, we directly assign the initial value and
```
Comment (evaliyev) [inline experimental/PropertyDDS/packages/property-properties/src/propertyFactory.js:1111]:
```suggestion
                // If there are no children, we directly assign the initial value and
```

File: experimental/PropertyDDS/packages/property-properties/src/propertyFactory.js [hunk 2]
```diff
@@ -1283,31 +1659,35 @@ class PropertyFactory {
...
-                            !!(templateOrConstructor.inherits), in_evaluateConstants);
+                        this._parseTemplate(templateOrConstructor, in_scope,
+                            !!(templateOrConstructor.inherits), out_propertyDef);
                     } else {
                         // If we have other contexts, we have to create the corresponding property object for that context
 
                         // check if a specialized collection is needed
                         var isEnum = this.inheritsFrom(typeid, 'Enum', { scope: in_scope });
 
                         var result;
```
Comment (evaliyev) [inline experimental/PropertyDDS/packages/property-properties/src/propertyFactory.js:1678]: unused?

File: experimental/PropertyDDS/packages/property-properties/src/propertyFactory.js [hunk 3]
```diff
@@ -1444,37 +1828,52 @@ class PropertyFactory {
...
-                    const constant = this._createFromPropertyDeclaration(constants[i], undefined, in_scope, true);
+
+                    const newChildEntry = {
+                        initialValue: undefined,
+                        constant: true
+                    };
+                    propertyDef.children = propertyDef.children || [];
+                    propertyDef.children.unshift([constants[i].id, newChildEntry]);
+
+                    this._createDefFromPropertyDeclaration(constants[i], in_scope, newChildEntry);
```
Comment (evaliyev) [inline experimental/PropertyDDS/packages/property-properties/src/propertyFactory.js:1872]: Function is a bit confusing. Sometimes it returns a result and sometimes relies on a side effect on one of the arguments.

File: experimental/PropertyDDS/packages/property-properties/src/propertyFactory.js [hunk 4]
```diff
@@ -1097,26 +1470,21 @@ class PropertyFactory {
     };
 
     /**
-     * Creates a property object that serves as parent for the template with the given typeid, when none has yet
-     * been created,
+     * Creates a property defintion for a non-collection property with the entry and constructor function assigned
```
Comment (evaliyev) [inline experimental/PropertyDDS/packages/property-properties/src/propertyFactory.js:1473]:
```suggestion
     * Creates a property definition for a non-collection property with the entry and constructor function assigned
```

**Review by evaliyev — APPROVED** [review]

File: experimental/PropertyDDS/packages/property-properties/src/properties/containerProperty.js [hunk 5]
```diff
@@ -24,9 +24,14 @@ export class ContainerProperty extends IndexedCollectionBaseProperty {
      * @protected
      */
     constructor(in_params) {
-        super({ typeid: 'ContainerProperty', ...in_params });
-        this._dynamicChildren = {};
-        this._optionalChildren = {};
+        super( in_params );
+        //this._dynamicChildren = {};
```
[the remark this answers is not in the document]
Comment (ruiterr) [reply experimental/PropertyDDS/packages/property-properties/src/properties/containerProperty.js:28]: yes

File: experimental/PropertyDDS/packages/property-properties/src/propertyFactory.js (hunk 3 above)
Comment (ruiterr) [reply experimental/PropertyDDS/packages/property-properties/src/propertyFactory.js:1872]: The function actually never returns anything any longer, but writes into the output parameter. The cases, where a return value was assigned, this was no used afterwards.

**Comment by tylerbutler** [comment]
@ruiterr Sorry for not merging this in sooner! It looks like a merge conflict needs to be resolved. Once that's done I'll merge ASAP. Please feel free to @-mention me to help with the mechanics of getting PRs merged.

**Review by DLehenbauer — APPROVED** [review]

**Pull request merged**